events/management/commands/rebuild_event_search_index.py

```diff
@@ -1,99 +1,29 @@
 import logging
 
-from django.contrib.postgres.search import SearchVector
 from django.core.management.base import BaseCommand
-from django.db.models import OuterRef, Subquery
-from django.utils import timezone
 
 from events.models import Event, EventSearchIndex
-from events.utils import get_field_attr, split_word_bases
+from events.search_index.postgres import EventSearchIndexService
 
 logger = logging.getLogger(__name__)
 
 
-def rebuild_search_index(lang: str = "fi") -> int:
+def rebuild_event_search_index() -> int:
     """
     Rebuild the full text search index for the events.
 
     This function iterates through all events, extracts words from the specified
     columns, and updates the EventSearchIndex model with data for
     the specified language.
 
-    :param lang: The language code for the search objects (default is "fi").
-    :return: The number of populated search objects.
+    :return: The number of updated search objects.
     """
-    num_populated = 0
+    num_updated = 0
     qs = Event.objects.all().order_by("pk")
```

Comment:
```suggestion
    qs = Event.objects.all().select_related("location").prefetch_related("keywords").order_by("pk")
```

probably speeds up things a little bit more

Reply:
Updated to PR.